### PR TITLE
Implement filter repository and tags by label API

### PR DESCRIPTION
**Pull request by ywk253100**

### CODE REVIEW

File: src/common/dao/repository_test.go [hunk 1]
```diff
@@ -32,61 +32,68 @@ var (
...
+	n, err := GetTotalOfRepositories()
+	require.Nil(t, err)
+	assert.Equal(t, total+1, n)
+}
 
-	exist := false
-	for _, repo := range repositories {
-		if repo.Name == name {
-			exist = true
+func TestGetRepositories(t *testing.T) {
```
Comment (reasonerjt) [inline src/common/dao/repository_test.go]: labelID is not covered.

File: src/common/dao/repository.go [hunk 2]
```diff
@@ -124,46 +103,79 @@ func GetTopRepos(projectIDs []int64, n int) ([]*models.RepoRecord, error) {
...
+
+	if len(q.ProjectName) > 0 {
+		sql += `and p.name = ? `
+		params = append(params, q.ProjectName)
 	}
-	_, err := qs.All(&repositories)
 
-	return repositories, err
+	if len(q.LabelIDs) > 0 {
+		sql += fmt.Sprintf(`and rl.label_id in ( %s ) `,
```
Comment (reasonerjt) [inline src/common/dao/repository.go]: does it mean the repo will be returned if it has one of the labels in the query parm?

File: src/common/models/repo.go [hunk 3]
```diff
@@ -37,3 +37,12 @@ type RepoRecord struct {
...
 	return RepoTable
 }
+
+// RepositoryQuery : query parameters for repository
+type RepositoryQuery struct {
+	Name        string
+	ProjectIDs  []int64
+	ProjectName string
+	LabelIDs    []int64
+	Pagination
```
Comment (reasonerjt) [inline src/common/models/repo.go]: The usage of `Pagination` in `RepositoryQuery` and `Label` is not consistent with other models, such as `project` `user`

File: src/ui/api/repository.go [hunk 4]
```diff
@@ -377,14 +382,19 @@ func (ra *RepositoryAPI) GetTag() {
 	}
 
 	result := assembleTags(client, repository, []string{tag},
-		ra.SecurityCtx.GetUsername())
+		ra.SecurityCtx.GetUsername(), nil)
```
Comment (reasonerjt) [inline src/ui/api/repository.go]: Why the labels is `nil`?

File: src/ui/api/statistic.go [hunk 5]
```diff
@@ -64,17 +64,22 @@ func (s *StatisticAPI) Get() {
 	}
 
 	statistic[PubPC] = (int64)(len(pubProjs))
```
Comment (reasonerjt) [inline src/ui/api/statistic.go]: This line should also be removed?

Comment (ywk253100) [reply src/ui/api/statistic.go]: It's the count of public projects, it's needed.

**Pull request merged**